### ISSUE TITLE
Script hangs on `set_seed` and `accelerator.backward(loss)` with A100 GPU

### ISSUE THREAD

**Issue by ProGamerGov** [huggingface/accelerate#788]
### System Info

```Shell
- `Accelerate` version: 0.12.0
- Platform: Linux-4.19.0-21-cloud-amd64-x86_64-with-debian-10.13
- Python version: 3.7.12
- Numpy version: 1.21.6
- PyTorch version (GPU?): 1.12.1 (True)
- `Accelerate` default config:
        - compute_environment: LOCAL_MACHINE
        - distributed_type: NO
        - mixed_precision: no
        - use_cpu: False
        - num_processes: 1
        - machine_rank: 0
        - num_machines: 1
        - main_process_ip: None
        - main_process_port: None
        - main_training_function: main
        - deepspeed_config: {}
        - fsdp_config: {}
        - downcast_bf16: no
```

I also tested using `Accelerate` version: 0.13.2 and can confirm the issue is still present.

```
OS: Debian GNU/Linux 10 (buster) (x86_64)
GCC version: (Debian 8.3.0-6) 8.3.0
Clang version: Could not collect
CMake version: version 3.24.1
Libc version: glibc-2.10

Python version: 3.7.12 | packaged by conda-forge | (default, Oct 26 2021, 06:08:53)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-4.19.0-20-cloud-amd64-x86_64-with-debian-10.13
Is CUDA available: True
CUDA runtime version: 11.3.109
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: NVIDIA A100-SXM4-40GB
Nvidia driver version: 470.57.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
```

I'm using the `Debian 10 based Deep Learning VM for PyTorch CPU/GPU with CUDA 11.3 M98` image on Google Cloud Compute.

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

```
pip install git+https://github.com/ShivamShrirao/diffusers.git
wget https://raw.githubusercontent.com/ShivamShrirao/diffusers/main/examples/dreambooth/requirements.txt
pip install -r requirements.txt

huggingface-cli login

accelerate config # Single GPU on Google Cloud Compute, No CPU, No half precision.

wget -q https://github.com/ShivamShrirao/diffusers/raw/main/examples/dreambooth/train_dreambooth.py

# Optional xformers
# pip install -q https://github.com/TheLastBen/fast-stable-diffusion/raw/main/precompiled/A100/xformers-0.0.13.dev0-py3-none-any.whl
```

Setup the dataset and then run a bash script pointing to it:

```
export MODEL_NAME="runwayml/stable-diffusion-v1-5"
export VAE_NAME="stabilityai/sd-vae-ft-mse"
export INSTANCE_DIR="concept_images"
export CLASS_DIR="class_reg_images"
export OUTPUT_DIR="path-to-save-model"

accelerate launch train_dreambooth.py \
  --pretrained_model_name_or_path=$MODEL_NAME \
  --pretrained_vae_name_or_path=$VAE_NAME \
  --instance_data_dir=$INSTANCE_DIR \
  --class_data_dir=$CLASS_DIR \
  --output_dir=$OUTPUT_DIR \
  --with_prior_preservation --prior_loss_weight=1.0 \
  --instance_prompt="a photo of sks <concept>" \
  --class_prompt="<concept>" \
  --save_sample_prompt="photo of sks <concept>" \
  --resolution=512 \
  --train_batch_size=1 \
  --gradient_accumulation_steps=1 \
  --learning_rate=5e-6 \
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --num_class_images=1290 \
  --save_interval=500 \
  --max_train_steps=25000 \
  --train_text_encoder \
  --mixed_precision="no" \
  --not_cache_latents
```

Use the `--seed=1337` to trigger the hang at the `set_seed` function.

### Expected behavior

```Shell
It should not hang.
```

Issue originally reported here: https://github.com/ShivamShrirao/diffusers/issues/60

**Comment by zetyquickly** [comment]
Also encountered this issue on GCP using Tesla T4

**Comment by unography** [comment]
I've the same issue on GCP - it's solved for now by using custom Docker image - `pytorch/pytorch:1.13.1-cuda11.6-cudnn8-runtime`, instead of using GCP's own image

**Comment by zetyquickly** [comment]
confirm, pytorch/pytorch images do work

**Comment by muellerzr** [comment]
Since this is a GCP image issue, for now let me know if it's okay to close this with the solution being use `pytorch/pytorch` images @ProGamerGov as that likely means it's not something coming from Accelerate

**Comment by zetyquickly** [comment]
I cannot confirm that, the underlying issue still exists and I assume that's in Accelerate, taking into account the fact that pytorch/pytorch image can be used successfully on the GCP's hardware. Can we further investigate what might be the root cause of the freeze

**Comment by muellerzr** [comment]
Yes, more debugging details would help me to investigate this further @zetyquickly

**Comment by zetyquickly** [comment]
Give a guidance please. For now the initial message of this issue is almost the same setup that I've have and the freeze is happening in `backward` according to pdb

**Comment by muellerzr** [comment]
I've spun up an A100, I'll be investigating and running this myself tomorrow. Just to be sure, you are also using the `Debian 10 based Deep Learning VM for PyTorch CPU/GPU with CUDA 11.3 M98` image @zetyquickly?

**Comment by zetyquickly** [comment]
[this image](https://console.cloud.google.com/gcr/images/deeplearning-platform-release/global/pytorch-gpu.1-13@sha256:0ff4fadfc5da720a115d67886741d6b437033cadf3e16d95a3e266d7b2bdc129/details)

**Comment by ProGamerGov** [comment]
I just had it occur with StableTuner using the c2-deeplearning-pytorch-1-12-cu113-v20230126-debian-10 image

**Comment by seeM** [comment]
Hey @muellerzr 😄  I'm having this issue with a Tesla T4 on GCP, with the default PyTorch image, VM created today: `Google, Debian 10 based Deep Learning VM for PyTorch CPU/GPU with CUDA 11.3, M103, Deep Learning VM Image with PyTorch 1.13 and fast.ai preinstalled.`

**Comment by muellerzr** [comment]
Hey @seeM (and others :) ) I'll be looking at this during the coming week :)

**Comment by seeM** [comment]
In case it's helpful, creating a new conda environment from scratch fixed this for me. I followed [this comment](https://github.com/ShivamShrirao/diffusers/issues/60#issuecomment-1294311477), but I changed to the official diffusers repo, set `cuda-toolkit=11.6` to match the image, and didn't install xformers.

**Comment by josephenguehard** [comment]
Hi,

I had the same issue, but it worked after uninstalling `torch-xla`:

```bash
pip uninstall torch-xla
```

**Comment by unography** [comment]
> Can others verify if they had torch-xla installed on their machines? That's amazing if so. Nice catch @josephenguehard, I'll check this out today as well on my a100 setup

Hi, yes. It worked for me after uninstalling `torch-xla`

I faced this same issue with pytorch-lightning as well
In pytorch-lightning, it was trying to pick TPU since xla was installed, I guess something similar might be happening here as well

Uninstalling the package fixed it

**Comment by muellerzr** [comment]
Thanks all for your help on this, #1472 *should* fix this up :)

**Comment by muellerzr** [comment]
Closed now with #1472, please reopen or ping me in this issue if it happens again!
